src/main/java/seedu/address/model/assignment/AssignmentList.java

```diff
@@ -162,123 +163,218 @@ public void generateSchedule(int numDays) {
+    public ArrayList<Float> allocateHours(ArrayList<Float> cumulativeHoursPerDayList, LocalDateTime deadline,
+                                          float hoursToBeAllocated, int noOfDaysBetween, LocalDateTime currDateTime) {
+
+        // Keeps track of the amount of time allocated to each day for this assignment.
+        ArrayList<Float> hoursPerDayList = new ArrayList<>();
 
         if (noOfDaysBetween == 0) {
             // Allocate only the remaining time the user has before the deadline
-            float diffInHours = java.time.Duration.between(dateTime, deadline).toHours();
-            hoursPerDayList.set(0, hoursPerDayList.get(0) + Math.min(diffInHours, estHours));
+            float diffInHours = Duration.between(currDateTime, deadline).toHours();
+            cumulativeHoursPerDayList.set(0, cumulativeHoursPerDayList.get(0)
+                + Math.min(diffInHours, hoursToBeAllocated));
+            hoursPerDayList.add(Math.min(diffInHours, hoursToBeAllocated));
+
+            return hoursPerDayList;
 
         } else if (noOfDaysBetween > 0) {
-            float[] currMinAndSecondMin = getMinAndSecondMin(hoursPerDayList, noOfDaysBetween);
-            float numHoursToDeadline = 0;
+            float[] currMinAndSecondMin = getMinAndSecondMin(cumulativeHoursPerDayList, noOfDaysBetween);
+
+            int currSize = cumulativeHoursPerDayList.size();
+
+            // Initialise hoursPerDayList
+            for (int i = 0; i < currSize; i++) {
+                hoursPerDayList.add((float) 0);
+            }
 
             // Allocate hours to new days first (if any)
-            int numNewDays = noOfDaysBetween - (hoursPerDayList.size() - 1);
+            int numNewDays = noOfDaysBetween - (currSize - 1);
 
             if (numNewDays != 0) {
-                float hoursToAdd = estHours / numNewDays;
+                float hoursToAdd = hoursToBeAllocated / numNewDays;
                 float hoursAdded = Math.min(currMinAndSecondMin[0], hoursToAdd);
 
                 for (int i = 0; i < numNewDays; i++) {
+                    cumulativeHoursPerDayList.add(hoursAdded);
                     hoursPerDayList.add(hoursAdded);
-                    estHours -= hoursAdded;
+                    hoursToBeAllocated -= hoursAdded;
                 }
-                numHoursToDeadline += hoursAdded;
             }
 
             // Allocate hours to days with the least amount of work allocated in increasing order
-            while (currMinAndSecondMin[0] != currMinAndSecondMin[1]) {
-                int daysWithMinHours = getMinDays(hoursPerDayList, noOfDaysBetween, currMinAndSecondMin[0]);
+            while (currMinAndSecondMin[0] != currMinAndSecondMin[1] && hoursToBeAllocated != 0) {
+                int daysWithMinHours = getMinDays(cumulativeHoursPerDayList, noOfDaysBetween, currMinAndSecondMin[0]);
                 float diffBetweenMinAndSecondMin = currMinAndSecondMin[1] - currMinAndSecondMin[0];
```

Comment:
Why are you finding the diff though? What is it for haha

Reply:
errrrr so that I don't 'over-allocate' time to the days with minimum number of hours O: